src/Autocomplete/src/AutocompleteResultsExecutor.php

```diff
@@ -23,7 +27,8 @@ final class AutocompleteResultsExecutor
 {
     public function __construct(
         private DoctrineRegistryWrapper $managerRegistry,
-        private ?Security $security = null
+        private PropertyAccessorInterface $propertyAccessor,
```

Comment:
To protect BC and trigger a deprecation, we can make the top of the class look something like this:

```php
final class AutocompleteResultsExecutor
{
    private PropertyAccessorInterface $propertyAccessor;
    private ?Security $security;

    public function __construct(
        private DoctrineRegistryWrapper $managerRegistry,
        $propertyAccessor,
        /* Security $security = null */
    ) {
        if ($propertyAccessor instanceof Security) {
            trigger_deprecation('symfony/ux-autocomplete', '2.8.0', 'Passing a "%s" instance as the second argument of "%s()" is deprecated, pass a "%s" instance instead.', Security::class, __METHOD__, PropertyAccessorInterface::class);
            $this->security = $propertyAccessor;
            $this->propertyAccessor = new PropertyAccessor();
        } else {
            $this->propertyAccessor = $propertyAccessor;
            $this->security = func_num_args() >= 3 ? func_get_arg(2) : null;
        }
    }
```